go/vt/callerid/callerid.go

```diff
@@ -90,30 +91,23 @@ func GetSubcomponent(ef *vtrpcpb.CallerID) string {
 // NewContext adds the provided EffectiveCallerID(vtrpcpb.CallerID) and ImmediateCallerID(querypb.VTGateCallerID)
 // into the Context
 func NewContext(ctx context.Context, ef *vtrpcpb.CallerID, im *querypb.VTGateCallerID) context.Context {
-	ctx = context.WithValue(
-		context.WithValue(ctx, effectiveCallerIDKey, ef),
-		immediateCallerIDKey,
-		im,
-	)
-	return ctx
+	return context.WithValue(ctx, callerIDContextKey{}, callerIDContext{ef, im})
 }
 
 // EffectiveCallerIDFromContext returns the EffectiveCallerID(vtrpcpb.CallerID)
 // stored in the Context, if any
 func EffectiveCallerIDFromContext(ctx context.Context) *vtrpcpb.CallerID {
-	ef, ok := ctx.Value(effectiveCallerIDKey).(*vtrpcpb.CallerID)
-	if ok && ef != nil {
-		return ef
+	if cid, ok := ctx.Value(callerIDContextKey{}).(callerIDContext); ok {
+		return cid.ef
 	}
 	return nil
 }
 
 // ImmediateCallerIDFromContext returns the ImmediateCallerID(querypb.VTGateCallerID)
 // stored in the Context, if any
 func ImmediateCallerIDFromContext(ctx context.Context) *querypb.VTGateCallerID {
-	im, ok := ctx.Value(immediateCallerIDKey).(*querypb.VTGateCallerID)
-	if ok && im != nil {
```

Comment:
Also removed the extraneous `!= nil` check, which wasn't actually free. But is entirely unnecessary to check. If it's nil, can simply return nil.